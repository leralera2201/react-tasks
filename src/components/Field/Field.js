import React from "react";
import { FormGroup, Input, Label } from "reactstrap";

const Field = ({ label, name, ...otherProps }) => {
  return (
    <FormGroup>
      <Label htmlFor={name}>{label}</Label>
      <Input id={name} placeholder={label} name={name} {...otherProps} />
    </FormGroup>
  );
};

export default Field;
