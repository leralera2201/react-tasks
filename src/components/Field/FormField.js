import React from 'react';
import { FormGroup, Input, Label } from 'reactstrap';
import { Field } from 'react-final-form';
import PropTypes from 'prop-types';
import Errors from "../Errors";

const FormField = ({ label, name, type }) => {
  return (
    <Field name={name}>
      {({ input, meta: {
          invalid, active, error, submitFailed, visited
      }}) => (
        <FormGroup>
          <Label htmlFor={name}>{label}</Label>
          <Input
            {...input}
            id={name}
            placeholder={label}
            name={name}
            type={type}
            invalid={invalid && !active && (submitFailed || visited)}
          />
          {invalid && !active && (submitFailed || visited) &&
              <Errors errors={Object.values(error)} />
          }
        </FormGroup>
      )}
    </Field>
  );
}

FormField.propTypes = {
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
};

export default FormField;
