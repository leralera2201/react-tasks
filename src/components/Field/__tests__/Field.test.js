import React from "react";
import { shallow } from "enzyme";

import Field from "../Field";
import toJson from "enzyme-to-json";

describe("Testing Field component", () => {
  const props = {
    type: "text",
    label: "label",
    name: "name",
    value: "value",
    onChange: jest.fn(),
  };

  it("renders correctly", () => {
    const wrapper = shallow(<Field {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
