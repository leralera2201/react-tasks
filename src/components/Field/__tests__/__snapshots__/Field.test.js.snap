// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Field component renders correctly 1`] = `
<FormGroup
  tag='div'
>
  <Label
    htmlFor='name'
    tag='label'
    widths={
      Array [
        'xs',
        'sm',
        'md',
        'lg',
        'xl',
      ]
    }
  >
    label
  </Label>
  <Input
    id='name'
    name='name'
    onChange={[MockFunction]}
    placeholder='label'
    type='text'
    value='value'
  />
</FormGroup>
`;
